examples/mappings/lib.rs

```diff
@@ -0,0 +1,281 @@
+            let caller = Self::env().caller();
+            self.balances.take(caller)
+        }
+    }
+
+    #[cfg(test)]
+    mod e2e_tests {
+        type E2EResult<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+        #[ink_e2e::test]
+        async fn insert_and_get_works(
+            mut client: ink_e2e::Client<C, E>,
+        ) -> E2EResult<()> {
+            // given
+            let constructor = mappings::constructors::new();
+            let contract_id = client
+                .instantiate(&mut ink_e2e::alice(), constructor, 0, None)
+                .await
+                .expect("instantiate failed")
+                .account_id;
+
+            // when
+            let insert = mappings::messages::insert_balance(1_000);
+            let size = client
+                .call(&mut ink_e2e::alice(), contract_id.clone(), insert, 0, None)
+                .await
+                .expect("call failed")
+                .value;
+
+            // then
+            let get = mappings::messages::get_balance();
+            let balance = client
+                .call(&mut ink_e2e::alice(), contract_id.clone(), get, 0, None)
+                .await
+                .expect("call failed")
+                .value;
+
+            assert!(size.is_none());
+            assert_eq!(balance, Some(1_000));
+
+            Ok(())
+        }
+
+        #[ink_e2e::test]
+        async fn insert_and_contains_works(
+            mut client: ink_e2e::Client<C, E>,
+        ) -> E2EResult<()> {
+            // given
+            let constructor = mappings::constructors::new();
+            let contract_id = client
+                .instantiate(&mut ink_e2e::bob(), constructor, 0, None)
+                .await
+                .expect("instantiate failed")
+                .account_id;
+
+            let insert = mappings::messages::insert_balance(2_000);
+            let _ = client
+                .call(&mut ink_e2e::bob(), contract_id.clone(), insert, 0, None)
+                .await
+                .expect("call failed");
+
+            // then
+            let contains = mappings::messages::contains_balance();
+            let is_there = client
+                .call(&mut ink_e2e::bob(), contract_id.clone(), contains, 0, None)
+                .await
+                .expect("call failed")
+                .value;
+
+            assert!(is_there);
+
+            Ok(())
+        }
+
+        #[ink_e2e::test]
+        async fn reinsert_works(mut client: ink_e2e::Client<C, E>) -> E2EResult<()> {
+            // given
+            let constructor = mappings::constructors::new();
+            let contract_id = client
+                .instantiate(&mut ink_e2e::charlie(), constructor, 0, None)
+                .await
+                .expect("instantiate failed")
+                .account_id;
+
+            // when
+            let first_insert = mappings::messages::insert_balance(1_000_000);
```

Comment:
Bit of a nitpick, but I'd keep all the `insert` amounts consistent between tests unless there's a reason for changing them. 

I have to do a double check each test to reason about why this is `X` instead of `Y`

Reply:
Yes, so we don't get race condition in transaction pool